app/src/main/java/com/example/postindustriaandroid/ui/MainActivity.kt

```diff
@@ -1,86 +1,147 @@
-        val text = input_text_ET.text.toString()
+    private fun saveLastSearch(searchText: String) {
+        SharedPrefsManager.saveHistory(searchText)
+        lifecycleScope.launch(Dispatchers.IO){
+            val user_id = db.userDao().getUser(SharedPrefsManager.getLogin()).id
+            db.historyDao().insert(HistoryEntity(0,searchText,user_id))
+        }
+    }
 
-        val gson = GsonBuilder()
-            .setLenient()
-            .create()
+    private fun toHistoryActivity(){
+        lifecycleScope.launch(Dispatchers.IO) {
+            val user: UserEntity = db.userDao().getUser(login = SharedPrefsManager.getLogin())
+            val intent = Intent(this@MainActivity, HistoryActivity::class.java)
+            intent.putExtra(WebViewActivity.USERID, user.id)
+
+            startActivity(intent)
+        }
+    }
+
+    private fun toFavoriteActivity(){
+        lifecycleScope.launch(Dispatchers.IO) {
+            val user: UserEntity = db.userDao().getUser(login = SharedPrefsManager.getLogin())
+            val intent = Intent(this@MainActivity, FavouritePhotoActivity::class.java)
+            intent.putExtra(WebViewActivity.USERID, user.id)
+
+            startActivity(intent)
+        }
+    }
+
+    private fun executeSearch(){
+        textForSearch = input_text_ET.text.toString()
+        val gson = GsonBuilder().setLenient().create()
 
         val retrofit = Retrofit.Builder()
             .baseUrl(PhotoRepository.BASE_URL)
             .addConverterFactory(GsonConverterFactory.create(gson))
             .build()
-
         val service = retrofit.create(PhotoRepository::class.java)
-        
+
         val data: MutableMap<String, String> = HashMap()
             data["method"] = PhotoRepository.API_METHOD
             data["api_key"] = PhotoRepository.API_KEY
             data["format"] = PhotoRepository.API_FORMAT
             data["nojsoncallback"] = PhotoRepository.NOJSONCALLBACK
-            data["text"] = text
+            data["text"] = textForSearch
         val call = service.getPhoto(data)
 
         call.enqueue(object : Callback<FlickrPhotoResponce> {
             override fun onResponse(call: Call<FlickrPhotoResponce>, response: Response<FlickrPhotoResponce>) {
-                Log.d(TAG, "onResponse: $response")
-                convertJsonToUrl(response.body()!!)
+                createCardsList(response.body()!!)
+                photo_cards_rv.adapter?.notifyDataSetChanged()
             }
+
             override fun onFailure(call: Call<FlickrPhotoResponce>, t: Throwable) {
-                Log.e(TAG, "onFailure: ", t)
             }
         })
     }
 
-    fun convertJsonToUrl(element: FlickrPhotoResponce){
-        var textToView = ""
-        val sizeOfList: Int = element.photos.photo.size
+   private fun createCardsList(element: FlickrPhotoResponce) {
+       cardsList.clear()
+       element.photos.photo.forEach {
+           val photoCard = FlickrPhotoCard("", "")
```

Comment:
А почему не в одну строку?
```suggestion
           val photoCard = FlickrPhotoCard(textForSearch, it.generateUrl())
```